pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/sink/PulsarSink.java

```diff
@@ -364,8 +364,8 @@ public void write(Record<T> record) {
         SinkRecord<T> sinkRecord = (SinkRecord<T>) record;
         TypedMessageBuilder<T> msg = pulsarSinkProcessor.newMessage(sinkRecord);
 
-        if (record.getKey().isPresent() && !(record.getSchema() instanceof KeyValueSchema &&
-                ((KeyValueSchema) record.getSchema()).getKeyValueEncodingType() == KeyValueEncodingType.SEPARATED)) {
+        if (record.getKey().isPresent() && !(record.getSchema() instanceof KeyValueSchemaImpl &&
```

Comment:
would it make sense to use the interface `KeyValueSchema` in `instanceof` checks?

Reply:
makes sense